wire-library/wire-runtime-swift/src/main/swift/JSONString.swift

```diff
@@ -66,8 +68,31 @@ public struct JSONString<T : Hashable> : Hashable, Codable {
                 try container.encode(String(value))
             }
         default:
-            fatalError("Unsupported type \(T.self)")
+            if T.self is Optional<Int64>.Type || T.self is Optional<UInt64>.Type {
+                var container = encoder.singleValueContainer()
+                try container.encodeNil()
+            } else {
+                fatalError("Unsupported type \(T.self)")
+            }
         }
     }
 
 }
+
+private extension Decoder {
+    func decodeStringTo<T, Integer: FixedWidthInteger>(_ type: Integer.Type) throws -> T {
```

Comment:
Naming nit: Move the "to" into the parameter name to be more Swifty, like `decodeString(to type: Int64.Type) throws -> Int64`

Reply:
Right